pkg/kubectl/cmd/set/set_resources.go

```diff
@@ -215,6 +217,15 @@ func (o *ResourcesOptions) Run() error {
 			}
 			return nil
 		})
+		if !transformed {
+			if isSelectedPrintMode(o.Output) {
```

Comment:
This construct feels awkward and is too verbose.  We generally never check errors to o.Out for these sorts of cases.  

I would prefer not to return early here.  Returning early creates another point of inconsistency between the normal case (changing the item) and no changes.  We should print all the items listed.

Reply:
this `transform` work for all resources not changed so that i can needn't do encode work.  and i understand `one object` or `item` be called `resources`.

and i feel print all items not changed not beautiful. if you keep your viewpoint. i 'd more like  when
resources is `one item` print `no resources changed` and `items` print `all resources not changed`

Reply:
@smarterclayton i removed ```isSelectedPrintMode()```, because i don't find check error in other command. 

@liggitt if we want add it, we should update all. so i remove it in this pr. Do you have some different advices about `no resources changed`. because i and smartclayton difference in opinion. wdyt about this?

Reply:
done